lib/paperclip/storage/ftp/server.rb

```diff
@@ -89,6 +112,22 @@ def mkdir_p(dirname)
           end
         end
 
+        def mktree(tree, base = "/")
+          return unless tree.any?
+          list = connection.nlst(base)
+          tree.reject{|k,_| list.include?(k)}.each do |directory, sub_directories|
+            begin
+              connection.mkdir(base + directory)
+            rescue Net::FTPPermError
+              # This error can be caused by an already existing directory,
+              # maybe it was created in the meantime.
+            end
+          end
+          tree.each do |directory, sub_directories|
+            mktree(sub_directories, base + directory + "/")
+          end
+        end
```

Comment:
I don't like keeping two methods (`mkdir_p` and `mktree`) that lead to the same end result. So I suggest to refactor the code to just use the new `mktree` method.